### PR TITLE
Final Student Teacher Course Request & View

### PR DESCRIPTION
**Pull request by TravisGM92**
## Pull request type
Please check the type of change your PR introduces:
- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please describe):

## Problem/feature

_What problem are you trying to solve?_
CoursesController was bloated! Methods too long

## Solution

_How did you solve the problem?_
Abstract away all the API calls into service and facade classes

## Other changes (e.g. bug fixes, UI tweaks, small refactors)

- Add encrypted key for Travis CI to run tests
- added edge case to teacher courses crud spec
- Merge branch 'teacher_request_and_view' of github.com:Classroom-Wars/… …
- Remove excess view
- Update for database matching
- Clean up file for static database
- Merge branch 'teacher_student_show' of github.com:Classroom-Wars/Fron… …
- Merge branch 'student_course_show' of github.com:Classroom-Wars/Front… …
- Update for merge
- Update omniauth for static database matching …
- Update facade id to integer …
- Update roles attribute to frontend files …
- Update student files for omniauth handshake and stubs …
- Update teacher files for omniauth handshake and stub …
- Add to student course show

## Checklist
- [x] Code compiles correctly
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] All tests passing
- [ ] No rubocop offenses
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by TravisGM92** [comment]
Should be OK to merge. Travis CI failed cause of the environment variable it looks like

**Review by coreycartercodes — APPROVED** [review]

File: app/controllers/students/courses_controller.rb [hunk 1]
```diff
@@ -57,13 +39,41 @@ def create
       response = conn("/api/v1/students/courses").post do |request|
         request.body = student_course_creation_params
       end
-      student_course_data = JSON.parse(response.body, symbolize_names: true)
+      JSON.parse(response.body, symbolize_names: true)
       redirect_to students_courses_path
     else
       # Sad Path
```
Comment (Callbritton) [inline app/controllers/students/courses_controller.rb:45]: Any ideas on what this sad path would look like?

**Comment by TravisGM92** [comment]
@coreycartercodes is this ready to merge?
